blocks/api/parser.js

```diff
@@ -181,6 +181,28 @@ export function createBlockWithFallback( name, innerHTML, attributes ) {
 		// as invalid, or future serialization attempt results in an error
 		block.originalContent = innerHTML;
 
+		// When a block is invalid, attempt to validate again using a supplied `deprecated` definition.
```

Comment:
Should we extract this logic to a separate function, or would it be possible to take advantage of recursion here in returning via `createBlockWithFallback` using the deprecated block attempt attributes in place of the original?

Reply:
Thanks for the suggestion Andrew, the code is better now. I preferred to avoid recursion because the logic is a bit simpler than `createBlockWithFallback` and it may include a specific "migrate" call in the future #3673